meson.build

```diff
@@ -206,6 +236,9 @@ if get_option('install_headers')
   install_headers('src/core/transfer_request.h', install_dir: prefix_inc)
   install_headers('src/core/agent_data.h', install_dir: prefix_inc)
   install_headers('src/infra/mem_section.h', install_dir: prefix_inc)
+  if ucx_gpu_device_api_available
+    install_headers('src/api/gpu/ucx/nixl_device.cuh', install_dir: prefix_inc + '/gpu/ucx')
+  endif
```

Comment:
IMO we can install it unconditionally, since the presence of ucx device api in when NIXL is compiled does not directly mean it will exist/not when user application is compiled